src/publicKeyProvider/AxiosKeyProvider.ts

```diff
@@ -0,0 +1,40 @@
+import { AxiosInstance, AxiosRequestConfig, default as axios } from 'axios';
+
+import { ConnectInstallKeysCdnUrl, EnvironmentType, KeyProvider } from './KeyProvider';
+
+export interface AxiosKeyProviderEnvironmentArgs {
+  environment?: EnvironmentType;
+  config?: AxiosRequestConfig; // Override default Axios configuration
+}
+
+function isAxiosInstance(obj: unknown): obj is AxiosInstance {
+  const aux = obj as { get?: unknown };
+  return Boolean(aux?.get);
+}
+
+export type AxiosKeyProviderArgs = AxiosKeyProviderEnvironmentArgs | AxiosInstance;
+
+/**
+ * Axios implementation that downloads the public key.
+ */
+export class AxiosKeyProvider implements KeyProvider {
+  client: AxiosInstance;
+
+  constructor(optionsOrClient?: AxiosKeyProviderArgs) {
+    if (isAxiosInstance(optionsOrClient)) {
+      this.client = optionsOrClient;
+    } else {
+      this.client = axios.create({
+        baseURL: ConnectInstallKeysCdnUrl[optionsOrClient?.environment || 'production'],
+        timeout: 3000,
+        responseType: 'text',
+        ...optionsOrClient?.config,
+      });
+    }
+  }
+
+  async get(kid: string): Promise<string> {
```

Comment:
I am curious why Typescript doesn't treat missing parameters in the interface implementation an error.

Reply:
Different from Java, where each overloaded method is "compiled" to a different entry in the class, JavaScript does not take the parameters into account. So there can only one method `get` and as long as the parameters here are callable with the interface arguments it works.